.github/workflows/apple-perf.yml

```diff
@@ -0,0 +1,313 @@
+
+          echo "models=$(echo $MODELS | jq -Rc 'split(",")')" >> $GITHUB_OUTPUT
+          MAPPED_ARNS_JSON=$(printf '%s\n' "${MAPPED_ARNS[@]}" | jq -R . | jq -s .)
+          echo "devices=$(echo "$MAPPED_ARNS_JSON" | jq -c .)" >> $GITHUB_OUTPUT
+          echo "delegates=$(echo $DELEGATES | jq -Rc 'split(",")')" >> $GITHUB_OUTPUT
+
+  export-models:
+    name: export-models
+    uses: pytorch/test-infra/.github/workflows/macos_job.yml@main
+    needs: set-parameters
+    strategy:
+      matrix:
+          model: ${{ fromJson(needs.set-parameters.outputs.models) }}
+          delegate: ${{ fromJson(needs.set-parameters.outputs.delegates) }}
+      fail-fast: false
+    with:
+      runner: macos-latest-xlarge
+      python-version: '3.11'
+      submodules: 'true'
+      timeout: 60
+      upload-artifact: ios-models
+      script: |
+        set -eux
+
+        echo "::group::Setting up CI environment"
+        .ci/scripts/setup-conda.sh
+
+        BUILD_TOOL=cmake
+        # Setup MacOS dependencies as there is no Docker support on MacOS atm
+        GITHUB_RUNNER=1 PYTHON_EXECUTABLE=python ${CONDA_RUN} --no-capture-output \
+          .ci/scripts/setup-macos.sh "${BUILD_TOOL}"
+
+        if [[ ${{ matrix.delegate }} == "qnn" ]]; then
+          PYTHON_EXECUTABLE=python ${CONDA_RUN} --no-capture-output bash \
+            .ci/scripts/setup-qnn-deps.sh
+          PYTHON_EXECUTABLE=python ${CONDA_RUN} --no-capture-output bash \
+            .ci/scripts/build-qnn-sdk.sh
+        fi
+
+        if [[ ${{ matrix.delegate }} == "coreml" ]]; then
+          PYTHON_EXECUTABLE=python ${CONDA_RUN} --no-capture-output \
+            backends/apple/coreml/scripts/install_requirements.sh
+        fi
+
+        if [[ ${{ matrix.delegate }} == "mps" ]]; then
+          PYTHON_EXECUTABLE=python ${CONDA_RUN} --no-capture-output \
+            backends/apple/mps/install_requirements.sh
+        fi
+
+        ARTIFACTS_DIR_NAME=artifacts-to-be-uploaded/${{ matrix.model }}_${{ matrix.delegate }}
+        echo "::endgroup::"
+
+        echo "::group::Exporting ${{ matrix.delegate }} model: ${{ matrix.model }}"
+        BUILD_MODE="cmake"
+        DTYPE="fp32"
+
+        if [[ ${{ matrix.model }} =~ ^stories* ]]; then
+          # Install requirements for export_llama
+          PYTHON_EXECUTABLE=python ${CONDA_RUN} --no-capture-output \
+            bash examples/models/llama2/install_requirements.sh
+
+          # Test llama2
+          if [[ ${{ matrix.delegate }} == "xnnpack" ]]; then
+            DELEGATE_CONFIG="xnnpack+custom+qe"
+          fi
+          PYTHON_EXECUTABLE=python ${CONDA_RUN} --no-capture-output \
+            bash .ci/scripts/test_llama.sh "${{ matrix.model }}" "${BUILD_MODE}" "${DTYPE}" "${DELEGATE_CONFIG}" "${ARTIFACTS_DIR_NAME}"
+        else
+          PYTHON_EXECUTABLE=python ${CONDA_RUN} --no-capture-output \
+            bash .ci/scripts/test.sh "${{ matrix.model }}" "${BUILD_MODE}" "${{ matrix.delegate }}" "${ARTIFACTS_DIR_NAME}"
```

Comment:
Neither coreml nor mps are yet supported in this script

Reply:
I see, let me put a `TODO` comment here for now